src/api/getCircularRoute.ts

```diff
@@ -0,0 +1,49 @@
+import { ConfigEnv } from "./env.config";
+
+export type ItineraryPoint = {
+    lat: number;
+    lon: number;
+    name: string; //can 
+};
+
+
+const itineraryPointToString = (itineraryPoint: ItineraryPoint) => {
+    let itineraryPointString = "";
+    //loop through itinerary points and add to string if name is empty omit it
+        if (itineraryPoint.name === "") {
+            itineraryPointString += `${itineraryPoint.lon},${itineraryPoint.lat}|`;
+        } else {
+            //url encode the name
+            const name = encodeURIComponent(itineraryPoint.name);
+            itineraryPointString += `${itineraryPoint.lon},${itineraryPoint.lat},${name}|`;
+        }
+    return itineraryPointString.slice(0, -1);
+}
+
+export const fetchCircularRouteCSMDuration = async (itineraryPoint: ItineraryPoint, duration: number) => {
+    const url = `https://www.cyclestreets.net/api/journey.json?key=${ConfigEnv.apiKey}&itinerarypoints=${itineraryPointToString(itineraryPoint)}&plan=leisure&duration=${duration}`;
+    const response = await fetch(url);
+    const data = await response.json();
+    return data;
+}
+
+export const fetchCircularRouteCSMDistance = async (itineraryPoint: ItineraryPoint, distance: number) => {
```

Comment:
To jest dokładnie to samo co wyżej tylko z parametrem distance. Najlepiej zrobić jedną funkcję przyjmującą argument distance albo duration :)